File: ops/frontend/src/types/session.ts
import type { EodDataset } from './eod'
import type { SystemEvent } from './event'

export interface TradingSession {
  sessionId: string
  machineId: string
  machine: string
  status: 'open' | 'closed'
  startedAt?: string | null
  endedAt?: string | null
  lastEventAt?: string | null
  eventCount: number
  tradeCount: number
}

export interface SessionDetail {
  session: TradingSession
  recentEvents: SystemEvent[]
  eodDatasets: EodDataset[]
}

File: ops/backend/app/schemas/session.py
"""Trading-session read models for the ops dashboard."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field

from app.schemas.eod import EodDatasetView
from app.schemas.event import SystemEvent


class SessionView(BaseModel):
    sessionId: str
    machineId: str
    machine: str
    status: Literal["open", "closed"]
    startedAt: str | None = None
    endedAt: str | None = None
    lastEventAt: str | None = None
    eventCount: int
    tradeCount: int


class SessionDetailView(BaseModel):
    session: SessionView
    recentEvents: list[SystemEvent] = Field(default_factory=list)
    eodDatasets: list[EodDatasetView] = Field(default_factory=list)
